checks.d/nagios_runner.py

```diff
@@ -0,0 +1,52 @@
+import subprocess
+import sys
+import time
+
+from checks import AgentCheck
+
+# Runs arbitrary commands, but exects that the STDOUT from the process will
+# begin with a sigil that denotes the sucess or failure of the check.
+# Config file looks like:
+# init_config:
+# # Not required for this check
+#
+# instances:
+#     - name: "some.check.name1"
+#       command: "/path/to/command with args"
+#     - name: "some.check.name2"
+#       command: "/path/to/command with args2"
+class NagiosRunner(AgentCheck):
+  def __init__(self, name, init_config, agentConfig):
+    AgentCheck.__init__(self, name, init_config, agentConfig)
+    self.last_ts = {}
+
+  def check(self, instance):
+
+    cmd = instance.get('command')
+    name = instance.get('name')
+
+    status = AgentCheck.UNKNOWN
+    try:
+      output = subprocess.check_output(cmd, stderr=subprocess.STDOUT, shell=True)
```

Comment:
I wonder if `check_output` has the same problem we saw with pipes, if it's the case we implemented that workaround f298832b2cfdcb8e5846d50b239d83bdd07a5469 which is now in the `utils.subprocess_output` module.